### ISSUE TITLE
Configurable fields

### ISSUE THREAD

**Issue by neamtua** [runlinenl/nova-profile-tool#2]
Hi, it would be great if you could make the fields configurable. For instance I don't use the base model with name and I use first_name and last_name fields and I need to change that.

**Comment by neamtua** [comment]
Perhaps the best way to do it would be to make a config file and define the Nova resource file to read the fields() function content and in the same config to define an array of field names that you do not wish to allow editing.
I'm not sure if this is the best approach because I haven't read too much into the nova source code but if you can easily render the fields the same way nova does and take into account the rules and everything defined in the resource then it would be awesome.

**Comment by nasrulhazim** [comment]
it's good to have simple configuration.

for instance, I don't want the user to update their email address.

**Comment by wemersonrv** [comment]
@neamtua i agree with you, if profile can reflect the entire model it would be amazing. 

But, in some cases there are some fields we don't want to change, in my specific case users can login using both **email** and **CPF** fields. CPF is an brazilian document number similar and it's unique for person. So i don't need to add it to allow users to change these two files. 

And the password, i believe @patrickbergman gives the solution: Do not update password when its empty.

And of course, other files like birthdate, nickname, address; etc. it would be nice to be in the profile too!

So a configuration file detailing the fields to be updated would be interesting.

**Comment by JeasonZuo** [comment]
You can add this function into User model to fix this issue
`public function setPasswordAttribute($value) {
    if (!Hash::check(null, $value)) {
        $this->attributes['password'] = $value;
    }
}`
